jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -479,24 +506,65 @@ def __init__(self, client: Client, **kwargs):
-    # to the mount_datums api. however, this is not that simple as the name
-    # can't simply be looked up in the datuminfo. we will need to have some
-    # mapping to track which files within a datum belong to which name.
+    def _populate_name_table(self, input: pps.Input):
+        """
+        Given the input spec, populates the bi-directional name to commit_uri
+        mapping of the commits mounted by the input.
+        """
+        if input.pfs:
+            repo = pfs.Repo(
+                name=input.pfs.repo,
+                project=pfs.Project(
+                    name=input.pfs.project if input.pfs.project else "default"
+                ),
+            )
+            branch = pfs.Branch(
+                repo=repo, name=input.pfs.branch if input.pfs.branch else "master"
+            )
+            commit_id = (
+                input.pfs.commit
+                if input.pfs.commit
+                else self._client.pfs.inspect_branch(branch=branch).head.id
+            )
+            commit = pfs.Commit(repo=repo, id=commit_id, branch=branch)
+            commit_uri = commit.as_uri()
+            if commit_uri in self._repo_names:
+                raise ValueError(
+                    "Loading multiple instances of the same commit is currently not supported in the extension"
+                )
+            name = (
+                f"{commit.repo.project.name}_{commit.repo.name}_{commit.branch.name}"
+                if input.pfs.name is None
+                else input.pfs.name
+            )
+            self._repo_names[commit_uri] = name
+            self._repo_names[name] = commit_uri
+        if input.join:
+            for i in input.join:
+                self._populate_name_table(input=i)
+        if input.group:
+            for i in input.group:
+                self._populate_name_table(input=i)
+        if input.cross:
+            for i in input.cross:
+                self._populate_name_table(input=i)
+
     def mount_datums(self, input_dict: dict):
-        input = pps.Input().from_dict(input_dict["input"])
-        self._input = input
-        self._datum_list = list(self._client.pps.list_datum(input=input))
-        self._datum_index = 0
-        self._mount_time = datetime.datetime.now()
-        if len(self._datum_list) == 0:
-            raise ValueError("input produced no datums to mount")
-        self._update_mount()
+        try:
+            input = pps.Input().from_dict(input_dict["input"])
+            self._input = input
+            self._datum_list = list(self._client.pps.list_datum(input=input))
+            self._datum_index = 0
+            self._mount_time = datetime.datetime.now()
+            self._populate_name_table(input=input)
+            if len(self._datum_list) == 0:
+                self._reset()
+                raise ValueError("input produced no datums to mount")
+            self._update_mount()
+        except Exception as e:
+            self._reset()
```

Comment:
Why the reset here? I'm guessing this is like an escape hatch for "something went wrong - abort entire operation"?

Reply:
that's right, we don't want a bad input or unexpected error to leave the file manager in a half baked state since it may cause some strangeness to be displayed to the user. instead, we just unmount everything in that case